obs/obs.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+	"os"
+)
+
+type Project struct {
+	XMLName      xml.Name     `json:"project" xml:"project"`
+	Name         string       `json:"name" xml:"name,attr"`
+	Title        *string      `json:"title" xml:"title,omitempty"`
+	Description  *string      `json:"description" xml:"description,omitempty"`
+	URL          *string      `json:"url" xml:"url,omitempty"`
```

Comment:
I changed my mind about these fields. Let's make this field non-pointer string and remove `omitempty`. We're actually going to get the same result.

Reply:
but how is this working internally then?

Reply:
I'm not sure I fully understand your question. Can you elaborate please?

Reply:
mhh, why is there need to remove the pointers and omitempty from here? (these fields are after all not mandatory, isn't it?)

Reply:
Because working with string pointers (or with any other pointer to a primitive type) is a little bit complicated and doesn't provide good user experience. When defining the struct, you have to do something like this:

```go
str := "My Project Name"
p := &Project{
    URL: &str,
}
```

Instead of:

```go
p := &Project{
    URL: "My Project Name",
}
```

When you have multiple strings, like in our case, it gets even worse.

Right now, if you completely omit the URL when defining a project, you'll get XML like this one:

```xml
<project>
    ...
</project>
```

If you remove pointers and `omitempty`, you'll get XML like this one:

```xml
<project>
    <url></url>
</project>
```

These two XMLs are practically the same. OBS is even going to remove the empty tags automatically. However, the user experience of using the library is much better in the second case (when not using pointers to primitive types).